sentinel-adapter/sentinel-zuul2-adapter/src/main/java/com/alibaba/csp/sentinel/adapter/gateway/zuul2/filters/inbound/SentinelZuulInboundFilter.java

```diff
@@ -0,0 +1,164 @@
+import com.alibaba.csp.sentinel.adapter.gateway.zuul2.api.matcher.HttpRequestMessageApiMatcher;
+import com.alibaba.csp.sentinel.adapter.gateway.zuul2.callback.ZuulGatewayCallbackManager;
+import com.alibaba.csp.sentinel.adapter.gateway.zuul2.constants.ZuulConstant;
+import com.alibaba.csp.sentinel.adapter.gateway.zuul2.filters.EntryHolder;
+import com.alibaba.csp.sentinel.adapter.gateway.zuul2.filters.endpoint.SentinelZuulEndpoint;
+import com.alibaba.csp.sentinel.context.ContextUtil;
+import com.alibaba.csp.sentinel.slots.block.BlockException;
+import com.alibaba.csp.sentinel.util.StringUtil;
+import com.netflix.zuul.context.SessionContext;
+import com.netflix.zuul.filters.http.HttpInboundFilter;
+import com.netflix.zuul.message.http.HttpRequestMessage;
+import rx.Observable;
+import rx.schedulers.Schedulers;
+
+import static com.alibaba.csp.sentinel.adapter.gateway.common.SentinelGatewayConstants.*;
+
+/**
+ * Zuul2 inboundFilter for Sentinel.
+ *
+ * @author wavesZh
+ */
+public class SentinelZuulInboundFilter extends HttpInboundFilter {
+
+    private static final String DEFAULT_BLOCK_ENDPOINT_NAME = SentinelZuulEndpoint.class.getCanonicalName();
+
+    private final int order;
+
+    private final String blockedEndpointName;
+    /**
+     * if executor is null, flow control action will do on I/O thread
+     */
+    private final Executor executor;
+    /**
+     * true if blocked but the rest of inbound filters will be skipped;
+     * false even if blocked, user can invoke other inbound filters by yourself.
+     */
+    private final boolean fastError;
+
+    private final GatewayParamParser<HttpRequestMessage> paramParser = new GatewayParamParser<>(
+            new HttpRequestMessageItemParser());
+
+    public SentinelZuulInboundFilter(int order) {
+        this(order, null);
+    }
+
+	public SentinelZuulInboundFilter(int order, Executor executor) {
+        this(order, DEFAULT_BLOCK_ENDPOINT_NAME, executor, true);
+	}
+
+    public SentinelZuulInboundFilter(int order, String blockedEndpointName, Executor executor, boolean fastError) {
+        this.order = order;
+		this.blockedEndpointName = blockedEndpointName;
+		this.executor = executor;
+		this.fastError = fastError;
+	}
+
+    @Override
+    public int filterOrder() {
+        return order;
+    }
+
+    @Override
+    public Observable<HttpRequestMessage> applyAsync(HttpRequestMessage request) {
+        if (executor != null) {
+            return Observable.just(request).subscribeOn(Schedulers.from(executor)).flatMap(this::apply);
+        } else {
+            return Observable.just(request).flatMap(this::apply);
+        }
+    }
+
+    private Observable<HttpRequestMessage> apply(HttpRequestMessage request) {
+        SessionContext context = request.getContext();
+        String origin = parseOrigin(request);
+        Deque<EntryHolder> holders = new ArrayDeque<>();
+        String routeId = (String) context.get(ZuulConstant.PROXY_ID_KEY);
```

Comment:
Is this the standard and universal key for routes of Zuul 2.x?

Reply:
That's not.The standard method is to call `com.netflix.zuul.context.SessionContext#getRouteHost`，and the key is "routeVIP".